### PR TITLE
BACK-82 

### PR DESCRIPTION
**Pull request by Manizha-khorram**
New Message Adapter && Partner Users Adapter

### CODE REVIEW

File: src/adapters/websocket-adapter.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+*const eventListeners{
+*   newMessage : message => 
+*   console.log(`New message recieved ${message}`)
+*}
+*newMessageAdapter(eventListeners)
+**/
+export const newMessageAdapter = async(eventListeners: EventListeners) => {
+    
+    //creates a new instance of the Socket.IO client that is connected to the server
+    const socket = io(baseURL);
```
Comment (jpremo) [inline src/adapters/websocket-adapter.ts:25]: Instead of baseURL doesn't the url need to be ws://localhost:8000 or ws://render.com?

**Review by jpremo — COMMENTED** [review]
I think the link can just be ws://localhost:8000. When I test this I get a CORS error. I think you can fix this using by adding this on the backend.

`const initiateSocket = (server) => {
    const io = socketio(server, {
        cors: {
            origin: 'http://localhost:3000',
            methods: ["GET", "POST"]
        }
    })`

You would need to dynamically substitute the origin url if using the production server though.

Along with this some form of authorization needs to be added to the client side websockets as is it causes an error on the backend in the middleware that crashes the server.

I made a frontend branch called BACK-82-test which you can use to test the websockets on the client side. Let me know if you run into issues fixing these problems.

**Comment by larasodjati** [comment]
@jpremo I tried to test it on branch BACK-82-test by adding newMessageAdapter inside useEffect and new div inside return() as below, all I see is Websocket Test on the browser, could you please let us know how to test this? Thank you.
```
useEffect(() => {
        newMessageAdapter({
            newMessage: (message) => {
              setMessage(message);
            }
          });
    }, [])
```
```
 <div>
         <h1>WebSocket Test</h1>
          <p>{message}</p>
  </div>
```

File: src/adapters/websocket-adapter.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
+
+import {io , Socket} from 'socket.io-client'
+import { getCookie, cookieName } from '../util/Authentication';
+const socketURL = window.location.hostname === 'localhost'? 'ws://localhost:8000': 'wss://shelf-share.onrender.com'
+
+const jwtCookie = getCookie(cookieName)as { token: string }
+const jwtToken = jwtCookie ? jwtCookie.token: null;
```
Comment (jpremo) [inline src/adapters/websocket-adapter.ts:7]: Can you get the cookie inside of the adapters? That way we always fetch the most up to date value.

**Review by jpremo — APPROVED** [review]
One small change, but other than that it looks good. Feel free to merge afterwards. Nice work!

**Pull request merged**